docs/data_driven_testing.adoc

```diff
@@ -311,6 +311,35 @@ You can use named deconstruction with nested data pipes, but only on the innermo
 include::{sourcedir}/datadriven/DataSpec.groovy[tag=nested-and-named-multi-variable-data-pipe]
 ----
 
+== Cross-multiplying Data Providers
+
+Two or more consecutive data providers, be it a data table or a data pipe, can also be combined
+using a cartesian product using the `combined:` label between them. The following will result
+in these four executed tests
+
+- `a feature (1 3 5)`
+- `a feature (2 3 5)`
+- `a feature (1 4 6)`
+- `a feature (2 4 6)`
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/datadriven/DataSpec.groovy[tag=multiple-tables-combined]
+----
+
+Exactly the same result can be achieved using
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/datadriven/DataSpec.groovy[tag=pipe-and-table-combined]
+----
+
+NOTE: Combining with a derived data variable (`x = ...`) makes no sense and thus is forbidden.
+
+NOTE: If a data table takes part in a cross-multiplication, accessing columns of previous
+      data tables would not behave in an intuitive way and thus is currently forbidden. Accessing
+      previous columns within one data table even works while taking part in a cross-multiplication.
+
```

Comment:
You should probably describe that you can also use multiple `combined:` after each other, if I understood the code correctly.

Reply:
You understood the code correctly. And the "or more" in "Two or more" right in the beginning of the paragraph was intended to document that fact. :-)

Two:
```groovy
where:
a << [1, 2]
combined:
b | c
3 | 4
5 | 6
```
or more:
```groovy
where:
a << [1, 2]
combined:
b | c
3 | 4
5 | 6
combined:
d | e
7 | 8
9 | 10
```
Don't you think it is clear enough?

Reply:
Maybe I am missing something, but here from the code review I can't see the sample for the "or more" case in the documentation.

The second sample in your comment would be great in the documentation as an additional sample.

Reply:
No, you didn't miss something.
The example was only here to show the difference between two and more, I expected the reader to make that transfer or just try it. :-D
I now changed the two examples a bit, so that one has two factors and one has three factors and also rewrote the test so that the list of tests is also taken from the execution and so that the test also ensures it is really the same result for both cases.